django/db/backends/postgresql/operations.py

```diff
@@ -303,6 +340,13 @@ def bulk_insert_sql(self, fields, placeholder_rows):
         values_sql = ", ".join("(%s)" % sql for sql in placeholder_rows_sql)
         return "VALUES " + values_sql
 
+    def adapt_integerfield_value(self, value, internal_type):
+        if value is None or hasattr(value, "resolve_expression"):
+            return value
+        if is_psycopg3:
+            return self.integerfield_type_map[internal_type](value)
+        return super().adapt_integerfield_value(value, internal_type)
```

Comment:
I think what Florian mean here is this method can be defined at class creation time instead of having a runtime check for `is_psycopg3`

```suggestion
    if is_psycopg3:
        def adapt_integerfield_value(self, value, internal_type):
            if value is None or hasattr(value, "resolve_expression"):
                return value
            return self.integerfield_type_map[internal_type](value)
```

Reply:
Yes good, updated :+1: